Source/courses.py

```diff
@@ -0,0 +1,75 @@
+# Author: Isa Blancett
+# Project: Interactive Programming
+# Date: 10.30.2017
+# Description: Class for reading in a csv file containing course information and
+# creating and instance for each course
+# Acknowledgements: Script structure is modeled after Transcriptase's Game
+# repository on Git
+
+import csv
+
+class Courses(object):
+    """ class for the different course options the user can choose from
+
+    contains:
+    label - 4 letter id
+    course - full title
+    description - in-depth summary of course
+    max - total number of levels
+    reqs - dependencies, '%s%s%s' % (effected level, needed course, needed level)
+    lvl - current level
+    order - order picked by user
+    """
+
+    def setup(self, config):
+
+        self.config = config
+        self.label = config['label']
+        self.name = config['course']
+        self.description = config['description']
+        self.max = config['max']
+        self.reqs = config['dependencies']
+        self.range = config['xrange']
+        self.lvl = 0
+        self.order = 0
+
+
+
+def create_course(config):
+    """ creates a single instance of a course given a config dictionary
+
+    returns: single course instance
+    """
+
+    new_course = Courses()
+    new_course.setup(config)
```

Comment:
Consider changing `Courses.setup` to `Courses.__init__`, and changing these two lines to `new_course == Courses(config)`.

The principles are:
* Code that requires the client to perform a sequence of calls in order is fragile.
* Reduce the amount of time that an object is in an unusable state. It doesn't make sense to use `new_course` after `Course()` and before `new_course.setup(…)`, so shrink the period during which the class is exposed in this state.
* Make it impossible (or harder) for the client of a class or module to mis-use it.

This principles all come to the same thing here; I'm listing them as separate principles because they're all good design principles, and they don't always apply to the same situations.

In this case, replacing `Courses.setup` by `Courses.__init__` allows you to remove `create_course` entirely, and replace calls to `create_course(config)` by `Courses(config)`.